### PR TITLE
feat: tag clean website desc., entity paragraph, and title

### PR DESCRIPTION
**Pull request by tianjianjiang**
close: #148
close: #151

### CODE REVIEW

**Review by manandey — APPROVED** [review]
LGTM. Thanks!

File: bsmetadata/preprocessing_utils.py [hunk 1]
```diff
@@ -625,11 +675,14 @@ def preprocess(self, examples: Dict[str, List]) -> Dict[str, List[OneToOneFeatur
...
 
 
-class TitlePreprocessor(MetadataPreprocessor):
+class TitlePreprocessor(MetadataTagger):
     """An exemplary metadata preprocessor for adding titles information."""
 
     def __init__(self, col_to_store_metadata="metadata", col_title="html_title") -> None:
         self.col_title = col_title
+        self.title_regex = re.compile(r"<title[^>]*>(.*)</title>")
+        # TODO: A stricter regex pattern for speed
```
Comment (tianjianjiang) [inline bsmetadata/preprocessing_utils.py:684]: @manandey FYI

File: bsmetadata/preprocessing_utils.py [hunk 2]
```diff
@@ -659,7 +712,9 @@ def preprocess(self, examples: Dict[str, List]) -> Dict[str, List[OneToOneFeatur
             if not example_title:
                 continue
             title = example_title[0]
-            title = re.search("<title>(.*)</title>", title)
+            # TODO: Trim string for speed
+
+            title = self.title_regex.search(title)
```
Comment (tianjianjiang) [inline bsmetadata/preprocessing_utils.py:717]: @manandey FYI

**Comment by shanyas10** [comment]
Thanks @tianjianjiang!

**Pull request merged**